### ISSUE TITLE
why training finetune_openomni cause OOM?

### ISSUE THREAD

**Issue by junwenxiong** [patrick-tssn/OpenOmniNexus#3]
Hi, 

I am trying to train  stage2 with finetue_openomni, using the  pretrained weight of stage1 with finetune_auditory. But i encounter the OOM when runing the stage2, the memory is increasing in the process of training.  I'm not sure about why this happens? The training data is the voiceassistant 400k.

**Comment by junwenxiong** [comment]
is there any clues about this problem? I met the same situation on  the qwen2 and llama3.

**Comment by patrick-tssn** [comment]
Could you please specify your hardware information?

**Comment by junwenxiong** [comment]
Here is my server and env info: 
GPU: 4x5880  Ada 48 GB
CUDA: 12.4
Pytorch: 2.5.0
Transformer: 4.44.0

**Comment by patrick-tssn** [comment]
I believe that the GPU memory might be the issue here, as the code was tested on 2/4 80G A800/A100.

**Comment by junwenxiong** [comment]
OK, the gpu memory is also increasing on A100 gpu,especially in the stage2? But i see, the llama-omni just used 4 L40 48GB. Why it currently not working on my GPUS?

**Comment by patrick-tssn** [comment]
llama-omni does not contain any vision encoder

**Comment by junwenxiong** [comment]
Yeah,I only use the speech encoder, without visual encoder, because of the speech 400k dataset. So it's strange that the training process is not same with llama-omni.

**Comment by patrick-tssn** [comment]
No, but this code will load the vision encoder anyway.

**Comment by junwenxiong** [comment]
OK, i set the batch size to 1,  now.  But how much epoches you used in stage1 and stage2, respectively?

**Comment by patrick-tssn** [comment]
stage 1 takes less than 1 epoch, and it depends on the available GPU quota lmao... stage 2 takes 4-5 epochs until the loss is stable. I believe the CTC loss is 'weaker' than the CE, thus it needs more epochs.

**Comment by junwenxiong** [comment]
I finally figure it out after several days.  The qwen2.5-3B and 400k are used for  whole two stages training. The stage1 is about 5 epochs on 4x5880 GPUs, which loss is shown below: 

![Image](https://github.com/user-attachments/assets/2e1b9564-cc7b-4bb5-a426-52d4873cc437)

Then, for stage2, i also trained the model for 5 epochs, but the batch-size is set to 1, avoiding OOM.  The loss is here: 

![Image](https://github.com/user-attachments/assets/37a48a31-ef30-4501-9ca7-b40a37e1f149)

The final generated speech is fine, it can answer and understand the meaning of the question. The overall quality of the speech is just good, still need some improvements.

Thanks a lot for providing this general framework.
